Content/NPCs/Dungeon/CrescentCaster.cs

```diff
@@ -0,0 +1,505 @@
+                    NPC.frameCounter = 0;
+                    yFrame = 0;
+                    break;
+                case 1: //Winding up to cast
+                    NPC.frameCounter++;
+
+                    if (NPC.frameCounter % 4 == 0)
+                    {
+                        yFrame++;
+                    }
+
+                    if (yFrame == 10)
+                    {
+                        yFrame = 0;
+                        xFrame = 2;
+                        NPC.frameCounter = 0;
+                    }
+
+                    break;
+                case 2: //Supporting
+                    SupportBehavior();
+                    break;
+            }
+        }
+
+        private void UpdateBolts()
+        {
+            foreach (CrescentCasterBolt bolt in Bolts)
+            {
+                bolt.resetCounter += bolt.resetCounterIncrement;
+                bolt.MidPointDirection *= 1.05f;
+
+                bolt.MidPoint += bolt.MidPointDirection;
+
+                bolt.UpdateTrailPoints();
+
+                if (aiCounter % aiCounterReset < aiCounterReset - 10 && xFrame == 2)
+                {
+                    if (bolt.fade < 1)
+                        bolt.fade += 0.1f;
+                }
+                else if (bolt.fade > 0)
+                    bolt.fade -= 0.1f;
+
+                foreach (Vector2 point in bolt.BezierPoints())
+                    Lighting.AddLight(point, Color.Cyan.ToVector3() * 0.3f * bolt.fade);
+
+            }
+
+            foreach (CrescentCasterBolt bolt in Bolts.ToArray())
+            {
+                if (bolt.resetCounter > 20 && !bolt.cloneCreated) //Change bolt curve
+                {
+                    bolt.cloneCreated = true;
+                    CreateBolt(bolt.TargetNPC);
+                }
+
+                if (bolt.resetCounter > 30 || !IsValidTarget(bolt.TargetNPC))
+                    Bolts.Remove(bolt);
+            }
+        }
+
+        private void CreateBolt(NPC other)
+        {
+            Vector2 directionTo = NPC.DirectionTo(other.Center);
+
+            Vector2 midPoint = CalculateMidpoint(other);
+            Bolts.Add(new CrescentCasterBolt(other, NPC, midPoint, Main.rand.NextFloat(2.5f) * NPC.DirectionTo(midPoint), Main.instance.GraphicsDevice));
+        }
+
+        private void SupportBehavior()
+        {
+            NPC.frameCounter++;
+
+            if (NPC.frameCounter % 4 == 0)
+            {
+                yFrame++;
+                yFrame %= 5;
+            }
+
+            if (aiCounter % aiCounterReset < aiCounterReset - 10)
+            {
+                if (starOpacity < 1)
+                    starOpacity += 0.1f;
+            }
+            else if (starOpacity > 0)
+                starOpacity -= 0.1f;
+
+            foreach (NPC supportTarget in supportTargets)
+            {
+                BarrierNPC barrierNPC = supportTarget.GetGlobalNPC<BarrierNPC>();
+
+                barrierNPC.MaxBarrier = 250;
```

Comment:
should we be hard setting to 250? what if an enemy already has max barrier, shouldn't we be adding?

Reply:
fixed

Reply:
fixed